runtime/src/append_vec.rs

```diff
@@ -132,12 +137,46 @@ impl AppendVec {
             map,
             // This mutex forces append to be single threaded, but concurrent with reads
             // See UNSAFE usage in `append_ptr`
-            append_offset: Mutex::new(0),
-            current_len: AtomicUsize::new(0),
+            append_offset: Mutex::new(initial_len),
+            current_len: AtomicUsize::new(initial_len),
             file_size: size as u64,
         }
     }
 
+    #[allow(clippy::mutex_atomic)]
```

Comment:
We can't avoid this clippy lint?

Reply:
I'd like to do so, but it seems we can't because this is just moved from [here](https://github.com/solana-labs/solana/pull/7373/files#diff-657d1e7648a9fb74cb1582a9c48beb4bL409), not added by this PR and nor addressed by this PR.